### DISCUSSION TITLE
Support arithmetic assignment operators in with expressions

### DISCUSSION THREAD

**Discussion by mariusGundersen** [dotnet/csharplang#5941] [Language Ideas] [answered]
It would be neat if there was a way to support arithmetic assignment operators, like `+=`, `--` and friends, in `with` expressions. For example:

```csharp
record Data(int X, int Y);

var data = new Data(0, 0);

data with {
  // today
  X = data.X + 1,
  // tomorrow
  Y++
}
```

**Comment by CyrusNajmabadi** [comment, marked as answer]
I'm championing that here: https://github.com/dotnet/csharplang/issues/5176

**Reply by mariusGundersen** [reply]
Great! Does that only deal with `+=` and `-=`, or with all the different assignment operators, like `/=`, `a++`, `++a` and `??=`?

**Reply by CyrusNajmabadi** [reply]
It would be any compound operation. It would not include ++ or --.

**Reply by FaustVX** [reply]
Why it would not include `++` or `--` ?

**Reply by CyrusNajmabadi** [reply]
Because those already have legal meaning in a collection initializer.  The language currently however only supports assignment operation for object-initializers.  So we can make compound initializers legal without issue.  As `++` is just shorthand for `+= 1`, it's not worth having to deal with the conflict that arises trying to support it.
